src/include/souffle/SignalHandler.h

```diff
@@ -150,20 +153,34 @@ class SignalHandler {
      * Signal handler for various types of signals.
      */
     static void handler(int signal) {
-        const char* msg = instance()->msg;
-        std::string error;
+        // Within these bounds there's basically fk all we safely call/use b/c signals are async on
```

Comment:
Can we mention this as a comment in the PR and remove it from the source code.

Reply:
My only objection to that is that is that signal-restrictions are rather esoteric and odds are anyone trotting into this area in the future might not know of them. Evidence: The original author obviously didn't know. I only knew because I both saw an occasional segfault and remembered an article on Win32 signals.

I'll add the comment to the PR & for the sake of time if you still rather see it expunged or otherwise edited feel free to layer on edits.

Reply:
No worries. Let's add it to the source code. 

If you don't mind, can we slighlty reword the comment; some people might get offended by the language - otherwise, I don't have any objections. 

Great that you fixed the signal code!